src/agent/api/messagelistener.ts

```diff
@@ -79,13 +83,28 @@ export class MessageListener extends events.EventEmitter {
                 console.error('Unauthorized.  Confirm credentials are correct and restart.  Exiting.');
                 return;
             }
-
+            
             if (err) {
                 onError(new Error('Could not create an agent session.  Retrying in ' + QUEUE_RETRY_DELAY/1000 + ' sec'));
                 onError(err);
-                setTimeout(() => {
-                        this.start(callback, onError);
-                    }, QUEUE_RETRY_DELAY);
+                
+                // retry 409 (session already exists) a few times
+                if (statusCode == 409) {
```

Comment:
Also note that the message listener implements event emmitter.  So, you can implement a giving up (think of better name) event then the agent can handle and put in it's log as well